scripts/west_commands/runners/nrfjprog.py

```diff
@@ -150,58 +151,124 @@ def get_board_snr(self, glob):
+
+        if self.family == 'NRF53':
+            if self.build_conf.get('CONFIG_SOC_NRF5340_CPUAPP', False):
+                self.coprocessor = 'CP_APPLICATION'
+            elif self.build_conf.get('CONFIG_SOC_NRF5340_CPUNET', False):
+                self.coprocessor = 'CP_NETWORK'
+            else:
+                raise RuntimeError(f'unknown nRF53; update {__file__}')
+            self.coprocessor_args = ['--coprocessor', self.coprocessor]
+        else:
+            self.coprocessor = None
+            self.coprocessor_args = []
+
+    def check_force_uicr(self):
+        # On SoCs without --sectoranduicrerase, we want to fail by
+        # default if the application contains UICR data and we're not sure
+        # that the flash will succeed.
+
+        # A map from SoCs which need this check to their UICR address
+        # ranges. If self.family isn't in here, do nothing.
+        uicr_ranges = {
+            'NRF53': ((0x00FF8000, 0x00FF8800),
+                      (0x01FF8000, 0x01FF8800)),
+            'NRF91': ((0x00FF8000, 0x00FF8800),),
+        }
+
+        if self.family not in uicr_ranges:
+            return
+
+        uicr = uicr_ranges[self.family]
+
+        if not self.force and has_region(uicr, self.hex_):
+            # Hex file has UICR contents.
+            raise RuntimeError(
+                'The hex file contains data placed in the UICR, which '
+                'needs a full erase before reprogramming. Run west '
+                'flash again with --force or --erase.')
+
+    def program_cmd(self, board_snr):
+        # Get the nrfjprog command use to actually program self.hex_.
+
+        # What type of erase argument should we pass to nrfjprog?
+        # The keys and values are:
+        # (family, coprocessor, --erase?): --nrfjprog-erase-argument
+        erase_args = {
+            ('NRF51', None, False): '--sectorerase',
+            ('NRF51', None, True): '--chiperase',
+
+            ('NRF52', None, False): '--sectoranduicrerase',
+            ('NRF52', None, True): '--chiperase',
+
+            # Mass erase with --recover is required to flash the
+            # network core using nrfjprog
+            ('NRF53', 'CP_NETWORK', True): '--recover',
+            ('NRF53', 'CP_APPLICATION', False): '--sectorerase',
+            ('NRF53', 'CP_APPLICATION', True): '--chiperase',
+
+            # This runner doesn't flash modem firmware, so we treat this
+            # as a single core SoC family.
+            ('NRF91', None, False): '--sectorerase',
+            ('NRF91', None, True): '--chiperase',
+        }
+
+        if self.family == 'NRF53':
+            if self.coprocessor == 'CP_NETWORK' and not self.erase:
+                raise RuntimeError(
+                    'please run "west flash --erase" instead; '
+                    'the nRF53 network core cannot be flashed '
+                    'independently and requires a full chip erase.')
```

Comment:
--recover is not needed for every programming sequence, just every programming sequence when there is no fw holding the chip open.